CORCI-668 build: Add Ubuntu packaging.

JohnMalmberg opened the pull request:
Create Ubuntu packaging based on exising Ubuntu package.

Signed-off-by: John.E.Malmberg <john.e.malmberg@intel.com>

Commit:
CORCI-668 build: Add Ubuntu packaging.

Create Ubuntu packaging based on exising Ubuntu package.

Signed-off-by: John.E.Malmberg <john.e.malmberg@intel.com>

brianjmurrell commented on Makefile:
```diff
@@ -1,81 +1,8 @@
 SRC_EXT := gz
 SOURCE   = https://github.com/01org/$(NAME)/archive/v$(VERSION).tar.$(SRC_EXT)
-PATCHES := $(NAME)-553f01f.patch
+#PATCHES := $(NAME)-553f01f.patch
```
Why are we commenting this out?

JohnMalmberg replied:
I commented it out during testing as it is a reverse patch.

When I went to prepare a forward patch, I discovered that DAOS was now specifying the release with out any patches.

brianjmurrell replied:
Ah, nice!.  So let's just delete the line, and the file.  #2 should have done that, but since it didn't this is a good opportunity to clean that up.

brianjmurrell commented on debian/control:
```diff
@@ -0,0 +1,58 @@
+ This package contains the development files needed to build against the shared
+ library.
+
+Package: libisal3
```
Why are we using libisal3 when the major version appears to be 2?

```
[2019-07-18T19:41:24.079Z] -rw-r--r-- root/root    257280 2019-06-27 14:41 ./usr/lib/x86_64-linux-gnu/libisal.so.2.0.26
```

JohnMalmberg replied:
This release removed a public symbol from the shared library with out bumping the soname,

I see though that future Ubuntu eoan code name has a 2.27.0-1 still using the libisal2 name.
Ubuntu 19.04 has has libisal2 2.25.

JohnMalmberg replied:
The future Ubuntu 2.27.0 just removed that symbol and kept the libname the same.
It also does not flag that it breaks the older releases.

brianjmurrell replied:
This seems like it's an upstream issue.  Hopefully they get better with their SONAME versioning.

Commit:
CORCI-668 build: remove unneeded patch

Removed the isa-l-553f01f.patch as DAOS is using the released version.

Signed-off-by: John.E.Malmberg <john.e.malmberg@intel.com>

Commit:
CORCI-668 build: Change to libisal2

The future Ubuntu upgrade to this package uses libisal2 and simply
deletes the symbol from the library, so doing the same here.

Signed-off-by: John.E.Malmberg <john.e.malmberg@intel.com>
